planner/core/plan_cost.go

```diff
@@ -39,27 +41,130 @@ func (p *basePhysicalPlan) GetPlanCost(taskType property.TaskType) (float64, err
 
 // GetPlanCost calculates the cost of the plan if it has not been calculated yet and returns the cost.
 func (p *PhysicalSelection) GetPlanCost(taskType property.TaskType) (float64, error) {
-	return 0, errors.New("not implemented")
+	if p.planCostInit {
+		return p.planCost, nil
+	}
+	var cpuFactor float64
+	switch taskType {
+	case property.RootTaskType, property.MppTaskType:
+		cpuFactor = p.ctx.GetSessionVars().CPUFactor
```

Comment:
So the calculation on TiDB and TiFlash share the same CPU factor?

Reply:
Currently Yes and it's not reasonable since execution engines on TiDB and TiFlash vary a lot. 
But this PR is only for refactoring so let's keep it unchanged. 
I'll introduce a new CPU factor for TiFlash in the next Sprint.